db-engine/src/main/java/com/splicemachine/db/impl/sql/compile/OptimizerFactoryImpl.java

```diff
@@ -52,12 +52,13 @@
 public class OptimizerFactoryImpl
 	implements ModuleControl, OptimizerFactory {
 
-	protected String optimizerId = null;
 	protected boolean ruleBasedOptimization = false;
 	protected boolean noTimeout = false;
 	protected boolean useStatistics = true;
 	protected int maxMemoryPerTable = Integer.MAX_VALUE;//may need to be long, but this also can be defined in configuration
 	protected long determineSparkRowThreshold; //The default value is defined in SQLConfiguration
+	protected int indexBatchSize;              //The default value is defined in SQLConfiguration
+	protected int indexLookupBlocks;           //The default value is defined in SQLConfiguration
```

Comment:
as far as i understand, this can be configured via `splice.index.batchSize` (default 4000) and `splice.index.numConcurrentLookups` (default 5), and these can't be configured by `SYSCS_SET_GLOBAL_DATABASE_PROPERTY`, but by site.xml ?
Do we have to change these settings or is this using the settings better?

Reply:
The idea is to consider these two configuration values in cost estimation. They decide how much parallelism we can have in executing index lookup, but we don’t estimate the cost accordingly. The goal of this change is to make index lookup costs reflect the same trend as real execution times.